articles/manual/type-model.md

```diff
@@ -0,0 +1,542 @@
+The subscripts of arrays are zero-based:
+
+```qsharp
+let arr = [10, 11, 36, 49];
+let ten = arr[0]; // 10
+let odds = arr[1..2..4]; // [11, 49]
+```
+
+## Tuple Types
+
+Given zero or more different types `T0`, `T1`, ..., `Tn`, we can denote a new  *tuple type* as `(T0, T1, ..., Tn)`.
+The types used to construct a new tuple type can themselves be tuples, as in `(Int, (Qubit, Qubit))`.
+Such nesting is always finite, however, as tuple types cannot under any circumstances contain themselves.
+
+Values of the new tuple type are tuples formed by sequences of values from each type in the tuple.
+For instance, `(3, false)` is a tuple whose type is the tuple type `(Int, Bool)`.
+It is possible to create arrays of tuples, tuples of arrays,
+tuples of sub-tuples, etc.
+
+As of Q# 0.3, `Unit` is the name of the *type*
+of the empty tuple; `()` is used for the empty tuple *value*.
+
+Tuple instances are immutable.
+Q# does not provide a mechanism to change the contents of a tuple
+once created.
+
+Tuples are a powerful concept used throughout Q# to collect values together into a single value, making it easier to pass them around.
+In particular, using tuple notation, we can express that every operation and callable takes exactly one input and returns exactly one output.
+
+### Singleton Tuple Equivalence
+
+It is possible to create a singleton (single-element) tuple, `('T1)`,
+such as `(5)` or `([1,2,3])`.
+However, Q# treats a singleton tuple as completely equivalent
+to a value of the enclosed type.
+That is, there is no difference between `5` and `(5)`, or between
+`5` and `(((5)))`, or between `(5, (6))` and `(5, 6)`.
+
+This equivalence applies for all purposes,
+including assignment and expressions.
+It is just as valid to write `(5)+3` as to write `5+3`,
+and both expressions will evaluate to `8`.
+In particular, this means that an operation or function whose input tuple or output tuple type has one field can be thought of as taking a single argument or returning a single value.
+
+We refer to this property as _singleton tuple equivalence_.
+
+## User-Defined Types
+
+A Q# file may define a new named type containing a single value of any legal type.
+For any tuple type `T`, we can declare a new user-defined type that is a subtype of `T` with the `newtype` statement.
+In the @"microsoft.quantum.canon" namespace, for instance, complex numbers are defined as a user-defined type:
+
+```qsharp
+newtype Complex = (Double, Double);
+```
+
+This statement creates a new type with two anonymous items of type `Double`.   
+Aside from anonymous items, user defined types also support named items as of Q# version 0.7 or higher. For example, we could have named the to items `Re` for the double representing the real part of a complex number and `Im` for the imaginary part: 
+
+```qsharp
+newtype Complex = (Re : Double, Im : Double);
```

Comment:
Similarly here, we may want to align with the actual definition of Microsoft.Quantum.Math.Complex.